napalm/junos/junos.py

```diff
@@ -235,14 +237,75 @@ def compare_config(self):
             return diff.strip()
 
     def commit_config(self, confirmed=None, message=None):
-        """Commit configuration."""
-        if confirmed is not None:
-            raise NotImplementedError
+        """
+        Commit configuration.
+
+        if confirmed is passed, random string is generated for the commit log
+        """
+        pending_commit_string = None
+        if confirmed:
+            pending_commit_string = str(uuid.uuid4()).split('-')[0]
```

Comment:
I find this strange. Why are you doing this?

Reply:
So I know the pending commit is mine. I can remove this and rely on the 'rollback' string of the commit message, however, I won't know for sure that rollback is from our `commit confirm`.